### PR TITLE
Cherrypick of #3797 to `release-5.13`

### PR DESCRIPTION
**Pull request by devinbinnie**
Cherrypick of #3797 to `release-5.13`

```release-note
NONE
```

## Change Impact: 🟡 Medium

**Regression Risk:** The changes introduce a size limit check to the widely-used `parseURL()` utility function, which is imported by 20+ files across the codebase (server info, diagnostics, plugins, views, windows, auth, and menus). While the 8KB limit is generous for standard URLs, the new validation creates a behavioral change: URLs exceeding this size now return `undefined` instead of attempting parsing. Callers already handle `undefined` returns, but any code assuming `parseURL` processes all valid URL strings without size restrictions could break. The check occurs before the existing try-catch block, so oversized URLs skip the normalization logic entirely. Risk is mitigated by the high threshold (8KB), which far exceeds typical URL lengths.

**QA Recommendation:** Light manual testing recommended. Focus on edge cases: (1) verify legitimate server URLs and navigation flows work normally, (2) test plugin URLs and custom endpoints to confirm none exceed 8KB, (3) validate that oversized URLs are properly rejected in permission dialogs and link handling. Automated test coverage is present (boundary condition tests added), but given the wide blast radius (20+ dependent modules) and critical nature of URL validation in navigation/security, manual verification of common navigation and authentication flows is prudent before release.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: src/common/utils/url.ts [hunk 1]
```diff
@@ -4,13 +4,18 @@
...
 
 export const getFormattedPathName = (pn: string) => (pn.endsWith('/') ? pn : `${pn}/`);
 export const parseURL = (inputURL: string | URL) => {
     if (inputURL instanceof URL) {
         return inputURL;
     }
+    const inputURLBytes = new TextEncoder().encode(inputURL).length;
+    if (inputURLBytes > MAX_URL_LENGTH) {
+        return undefined;
+    }
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/common/utils/url.ts:18]: We're only needing to check for URLs that pass in a string, we can assume that if the URL has been parsed already that it's already passed this check. The above type check is for redundancy.
